browser/brave_stats_updater_params.cc

```diff
@@ -70,6 +78,16 @@ void BraveStatsUpdaterParams::LoadPrefs() {
   week_of_installation_ = pref_service_->GetString(kWeekOfInstallation);
   if (week_of_installation_.empty())
     week_of_installation_ = GetLastMondayAsYMD();
+  date_of_installation_ = pref_service_->GetTime(kDateOfInstallation);
+  if (date_of_installation_.is_null() && GetFirstRun()) {
+    // We need to save the time here to handle the edge case when the referral
+    // ping is unsuccessful, and IsChromeFirstRun() will never be set again.
+    date_of_installation_ = GetCurrentTimeNow();
+    pref_service_->SetTime(kDateOfInstallation, date_of_installation_);
+  } else if (!date_of_installation_.is_null() &&
+        (GetCurrentTimeNow() - date_of_installation_ >= g_dtoi_delete_delta)) {
+      date_of_installation_ = base::Time();
```

Comment:
Just double-checking: intention here is to null out the time, right?

Reply:
Yep